src/server/game/Entities/Creature/Creature.cpp

```diff
@@ -2500,6 +2500,9 @@ std::string Creature::GetScriptName() const
 
 uint32 Creature::GetScriptId() const
 {
+    if (CreatureData const* creatureData = GetCreatureData())
```

Comment:
It's never null for db spawns, but not all are db spawns. Battlegrounds create creatures without creature data.
And yes it is hiding intentionally hiding template script id :)